tests/frontend/specs/helper.js

```diff
@@ -339,4 +400,66 @@ describe("the test helper", function(){
       done();
     });
   });
+
+  describe('helper',function(){
+    before(function(cb){
+      helper.newPad(function(){
+        cb();
+      })
+    })
+
+    it(".textLines() returns the text of the pad as strings", async function(){
+      let lines = helper.textLines();
+      let defaultText = helper.defaultText();
+      expect(lines).to.be.an('object');
+      expect(lines[0]).to.be.an('string');
+      // @todo
+      // final "\n" is added automatically, but my understanding is this should happen
+      // only when the default text does not end with "\n" already
+      expect(lines.join("\n")+"\n").to.equal(defaultText);
+    })
+
+    it(".linesDiv() returns the text of the pad as div elements", async function(){
+      let lines = helper.linesDiv();
+      let defaultText = helper.defaultText();
+      expect(lines).to.be.an('object');
+      expect(lines[0]).to.be.an('object');
+      expect(lines[0].text()).to.be.an('string');
+      _.map(defaultText.split("\n"), function(line, index){
+        // @todo last newline
+        if(!index === lines.length){
+          expect(lines[index].text()).to.equal(line);
+        }
+      })
+    })
+
+    it(".edit() defaults to send an edit to the first line", async function(){
+      let firstLine = helper.textLines()[0];
+      await helper.edit("line")
+      expect(helper.textLines()[0]).to.be(`line${firstLine}`);
+    })
+
+    it(".edit() to the line specified with parameter lineNo", async function(){
```

Comment:
Throws. The edit method is brittle right now, because it adds content where the caret is, without taking the caret position into account via parameter, but for the current tests it's probably enough.